CleverTapSDK/CTUserInfoMigrator.m

```diff
@@ -0,0 +1,46 @@
+//
+//  CTUserInfoMigrator.m
+//
+//  Created by Kushagra Mishra on 29/05/24.
+//
+
+#import "CTUserInfoMigrator.h"
+
+@implementation CTUserInfoMigrator
+
++ (void)migrateUserInfoFileForAccountID:(NSString *)acc_id deviceID:(NSString *)device_id {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    
+    // Get the path to the Documents directory
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSUserDomainMask, YES);
+    NSString *libraryPath = [paths objectAtIndex:0];
+    
+    // Construct the old plist file name and path
+    NSString *userProfileWithDeviceID = [NSString stringWithFormat:@"clevertap-%@-userprofile.plist", acc_id];
+    NSString *userProfileWithDeviceIDPath = [libraryPath stringByAppendingPathComponent:userProfileWithDeviceID];
+        
+    // Construct the new plist file name and path
+    NSString *userProfileWithDeviceIDAndGUID = [NSString stringWithFormat:@"clevertap-%@-%@-userprofile.plist", acc_id, device_id];
+    NSString *userProfileWithDeviceIDAndGUIDPath = [libraryPath stringByAppendingPathComponent:userProfileWithDeviceIDAndGUID];
+    
+    NSError *error = nil;
+    if ([fileManager fileExistsAtPath:userProfileWithDeviceIDAndGUIDPath]) {
```

Comment:
Check if old file exists, migrate and delete it. This will make one call to fileExists.

Reply:
Changes done